controllers/vspherecluster_controller.go

```diff
@@ -220,6 +220,12 @@ func (r clusterReconciler) reconcileDelete(ctx *context.ClusterContext) (reconci
 	if err != nil {
 		return reconcile.Result{}, err
 	}
+
+	if len(vsphereMachines) > 0 {
```

Comment:
@yastij 
Background question. Is the VM backing the HAProxy included in `vsphereMachines`? if so this will hang indefinitely 😅

Reply:
@fabriziopandini - no it's not included, `VSphereMachines` represent only Cluster API infra machines, anything that needs a VM generates a `VSphereVM`, so this check shouldn't block

Reply:
thanks for the clarification!